Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp

```diff
@@ -56,18 +56,19 @@ bool SystemSettingsManagerProxy::darkMode() const
         return true;
 
     // FIXME: These are just heuristics, we should get the dark mode from libhandy/libadwaita, falling back to the settings portal.
-
-    if (auto* themeNameEnv = g_getenv("GTK_THEME"))
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-        return g_str_has_suffix(themeNameEnv, "-dark") || g_str_has_suffix(themeNameEnv, "-Dark") || g_str_has_suffix(themeNameEnv, ":dark");
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+    // Or maybe just use the settings portal, because we don't want to depend on libhandy, and maybe don't want to depend on libadwaita?
```

Comment:
> // Or maybe just use the settings portal, because we don't want to depend on libhandy, and maybe don't want to depend on libadwaita?

Also just to answer this. libadwaita has its own internal state that can override the system setting, so you have to check it directly to truly match the app.

Reply:
I have code for this here, not too crazy: https://github.com/TingPing/WebKit/commit/e262dc251e5dc82493124751e64e0b57557e0a55